model/src/main/kotlin/PackageProvider.kt

```diff
@@ -0,0 +1,147 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+/**
+ * The provider / hosting platform of a remote artifact or Version Control System. These are basically human-readable
+ * aliases for different URLs that may point to the same provider. Note that different providers might host different
+ * artifacts under the same [Identifier]. So if in question, the provider must be given in addition to the [Identifier].
+ */
+enum class PackageProvider(
+    /**
+     * The regular expression patterns that match all URLs for a specific provider.
+     */
+    vararg urlPatterns: String,
+
+    /**
+     * The name ClearlyDefined uses for this provider. If a provider if not available in ClearlyDefined (yet), use an
+     * empty string.
+     */
+    val clearlyDefinedName: String = ""
+) {
+    COCOAPODS(
+        "^https?://cocoapods\\.org/pods/",
+        clearlyDefinedName = "cocoapods"
+    ),
+    CRATES_IO(
+        "^https?://crates\\.io/api/",
+        clearlyDefinedName = "cratesio"
+    ),
+    DEBIAN(
+        "^https?://(\\w+\\.)+debian.org/debian/",
+        clearlyDefinedName = "debian"
+    ),
+    GITHUB(
+        "^(https?|ssh)://(git@)?(\\w+\\.)?github\\.com/",
+        clearlyDefinedName = "github"
+    ),
+    GITLAB(
+        "^(https?|ssh)://(git@)?(\\w+\\.)?gitlab\\.com/",
+        clearlyDefinedName = "gitlab"
+    ),
+    GOLANG(
+        "^https?://go\\.googlesource\\.com/",
+        "^https?://proxy\\.golang\\.org/",
+        clearlyDefinedName = "golang"
+    ),
+    GRADLE_PLUGIN(
+        "^https?://plugins\\.gradle\\.org/m2/",
+        clearlyDefinedName = "gradleplugin"
+    ),
+    MAVEN_CENTRAL(
+        "^https?://repo\\.maven\\.apache\\.org/maven2/",
+        "^https?://repo1\\.maven\\.org/maven2/",
+        clearlyDefinedName = "mavencentral"
+    ),
+    MAVEN_GOOGLE(
+        "^https?://maven\\.google\\.com/",
+        clearlyDefinedName = "mavengoogle"
+    ),
+    NPM_JS(
+        "^https?://registry\\.npmjs\\.(com|org)/",
+        clearlyDefinedName = "npmjs"
+    ),
+    NUGET(
+        "^https?://api\\.nuget\\.org/.+\\.nupkg",
+        clearlyDefinedName = "nuget"
+    ),
+    PACKAGIST(
+        "^https?://repo\\.packagist\\.org/",
+        clearlyDefinedName = "packagist"
+    ),
+    PYPI(
+        "^https?://files\\.pythonhosted\\.org/packages/",
+        clearlyDefinedName = "pypi"
+    ),
+    RUBYGEMS(
+        "^https?://rubygems\\.org/gems/",
+        clearlyDefinedName = "rubygems"
+    ),
+
+    // The following have no matching ClearlyDefined provider.
+    ANDROID(
+        "^https?://android\\.googlesource\\.com/"
+    ),
+    APACHE_GIT(
+        "^https?://gitbox\\.apache\\.org/repos/.+\\.git",
+        "^https?://git-wip-us\\.apache\\.org/repos/.+\\.git"
```

Comment:
I fail to understand the point of `.+`. Could you help?

Reply:
I want to ensure that the URLs end with ".git", and that something comes between that and the base URL / base path.